src/main/java/org/sunbird/course/service/ExploreCourseServiceImpl.java

```diff
@@ -183,4 +185,135 @@ public SBApiResponse getExploreCourseListV2() {
 	}
+
+	@Override
+	public SBApiResponse upsertExploreCourse(Map<String, Object> requestObj) {
+		SBApiResponse response =
+				ProjectUtil.createDefaultResponse(Constants.API_EXPLORE_COURSE_UPDATE);
+		logger.info("ExploreCourseService::upsertExploreCourse:inside method");
+		Map<String, Object> masterData = (Map<String, Object>) requestObj.get(Constants.REQUEST);
+		String errMsg = validateUpsertRequest(masterData);
+		if (!StringUtils.isEmpty(errMsg)) {
+			response.getParams().setErrmsg(errMsg);
+			response.setResponseCode(HttpStatus.BAD_REQUEST);
+			return response;
+		}
+		try {
+			List<?> dataList = (List<?>) masterData.get(Constants.DATA);
+			Iterator<?> iterator = dataList.iterator();
+
+			while (iterator.hasNext()) {
+				Object item = iterator.next();
+				Map<?, ?> itemMap = (Map<?, ?>) item;
+
+				// Check for "identifier" key
+				Map<String, Object> request = new HashMap<>();
+				request.put(Constants.IDENTIFIER, itemMap.get(Constants.IDENTIFIER));
+
+				List<Map<String, Object>> listOfMasterData = cassandraOperation.getRecordsByProperties(
+						Constants.KEYSPACE_SUNBIRD, Constants.TABLE_EXPLORE_COURSE_LIST_V2, request,
+						new ArrayList<>());
+
+				if (!CollectionUtils.isEmpty(listOfMasterData)) {
+					Map<String, Object> updateRequest = new HashMap<>();
+					updateRequest.put(Constants.SEQUENCE_NO, itemMap.get(Constants.SEQUENCE_NO));
+					Map<String, Object> updateResponse = cassandraOperation.updateRecord(
+							Constants.KEYSPACE_SUNBIRD, Constants.TABLE_EXPLORE_COURSE_LIST_V2, updateRequest,
+							request);
+
+					if (updateResponse != null && !Constants.SUCCESS.equalsIgnoreCase(
+							(String) updateResponse.get(Constants.RESPONSE))) {
+						errMsg = String.format("Failed to update details");
+						response.getParams().setErrmsg(errMsg);
+						response.setResponseCode(HttpStatus.BAD_REQUEST);
+						return response;
+					}
+				} else {
+					request.put(Constants.SEQUENCE_NO, itemMap.get(Constants.SEQUENCE_NO));
+					response = cassandraOperation.insertRecord(Constants.KEYSPACE_SUNBIRD,
+							Constants.TABLE_EXPLORE_COURSE_LIST_V2, request);
+					if (!Constants.SUCCESS.equalsIgnoreCase((String) response.get(Constants.RESPONSE))) {
+						errMsg = String.format("Failed to create position");
+						response.setResponseCode(HttpStatus.BAD_REQUEST);
+						response.getParams().setErrmsg(errMsg);
+						return response;
+					}
+				}
+			}
+
+		} catch (Exception e) {
+			errMsg = String.format("Exception occurred while performing upsert operation");
+			logger.error(errMsg, e);
+		}
+		if (org.apache.commons.lang.StringUtils.isNotBlank(errMsg)) {
+			response.getParams().setStatus(Constants.FAILED);
+			response.getParams().setErrmsg(errMsg);
+			response.setResponseCode(HttpStatus.BAD_REQUEST);
+			return response;
+		}
+		response.setResponseCode(HttpStatus.OK);
+		response.getResult().put(Constants.RESPONSE, Constants.SUCCESS);
+		response.getParams().setStatus(Constants.SUCCESS);
```

Comment:
No need of these -- the createDefaultResponse method should set the value to success by default.
And avoid setting success status from below code.

Reply:
made changes accordingly